### PR TITLE
Remove insights

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Removing Insights from Mattermost server and webapp codebase.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53431

#### Release Note
```release-note
Removed deprecated Insights feature.
```

### CODE REVIEW

**Comment by harshilsharma63** [comment]
@marianunez TODO items here-

1. Update E2E tests as some are failing. Its easy as we just either need to delete the insight tests or remove the part checking for insight on the LHS.
2. Removing insights code from Playbooks and Boards. Its fine for v8.1 as that code only executes if the corresponding plugin hooks exist in client side, and I've removed those in this PR.

**Comment by marianunez** [comment]
> @marianunez TODO items here-
> 
> 1. Update E2E tests as some are failing. Its easy as we just either need to delete the insight tests or remove the part checking for insight on the LHS.
> 2. Removing insights code from Playbooks and Boards. Its fine for v8.1 as that code only executes if the corresponding plugin hooks exist in client side, and I've removed those in this PR.

@harshilsharma63 We'll fix the E2E tests in this PR but for 2. can we create a follow up ticket for the clean up on the Boards and Playbooks and if you have them readably available which endpoints would need to be removed?

File: server/public/model/insights.go [hunk 1]
```diff
@@ -268,6 +268,7 @@ func StartOfDayForTimeRange(timeRange string, location *time.Location) *time.Tim
 // GetStartOfDayForTimeRange gets the unix start time in milliseconds from the given time range.
 // Time range can be one of: "today", "7_day", or "28_day".
 func GetStartOfDayForTimeRange(timeRange string, location *time.Location) (*time.Time, *AppError) {
```
Comment (marianunez) [inline server/public/model/insights.go:270]: Shouldn't we remove this whole model file? Wondering why Harshill didn't remove it

File: webapp/channels/src/components/channel_layout/center_channel/center_channel.tsx [hunk 2]
```diff
@@ -129,12 +119,7 @@ export default class CenterChannel extends React.PureComponent<Props, State> {
                             path='/:team/drafts'
                             component={LazyDrafts}
                         />
-                        {insightsAreEnabled ? (
-                            <Route
-                                path='/:team/activity-and-insights'
-                                component={LazyActivityAndInsights}
-                            />
-                        ) : null}
+
```
Comment (larkox) [inline webapp/channels/src/components/channel_layout/center_channel/center_channel.tsx:122]: Unneeded line break?

File: webapp/channels/src/packages/mattermost-redux/src/utils/channel_utils.ts [hunk 3]
```diff
@@ -367,6 +367,7 @@ export function calculateUnreadCount(
     member: ChannelMembership | null | undefined,
     crtEnabled: boolean,
 ): {showUnread: boolean; mentions: number; messages: number; hasUrgent: boolean} {
+    // LOL
```
Comment (larkox) [inline webapp/channels/src/packages/mattermost-redux/src/utils/channel_utils.ts:370]: LOL?

File: webapp/platform/types/src/insights.ts [hunk 4]
```diff
@@ -69,6 +69,7 @@ export type TopChannelActionResult = {
     error?: any;
 };
 
+// LOL
```
Comment (larkox) [inline webapp/platform/types/src/insights.ts:72]: LOL? Shouldn't this whole file be deleted?

**Review by larkox — CHANGES_REQUESTED** [review]
Some minor changes.

File: webapp/channels/src/packages/mattermost-redux/src/utils/channel_utils.ts (hunk 3 above)
Comment (marianunez) [reply webapp/channels/src/packages/mattermost-redux/src/utils/channel_utils.ts:370]: Not sure what this is about - maybe he left that as a reminder to delete? I noticed the same here: https://github.com/mattermost/mattermost/pull/23952#discussion_r1255016576

Comment (marianunez) [reply webapp/channels/src/packages/mattermost-redux/src/utils/channel_utils.ts:370]: I'll just delete this pending files (I'm working on this PR as Harshil is out)

Comment (marianunez) [reply webapp/channels/src/packages/mattermost-redux/src/utils/channel_utils.ts:370]: This one is related to unreads so I don't think we need to delete the whole file. I'll just remove the comment.

File: webapp/platform/types/src/insights.ts (hunk 4 above)
Comment (marianunez) [reply webapp/platform/types/src/insights.ts:72]: Not sure what this is about - maybe he left that as a reminder to delete? I noticed the same here: https://github.com/mattermost/mattermost/pull/23952#discussion_r1255016576

File: e2e-tests/cypress/tests/integration/channels/message_forwarding/forward_message_from_dm_spec.js [hunk 5]
```diff
@@ -25,11 +25,8 @@ describe('Forward Message', () => {
     const commentMessage = 'Comment for the forwarded message';
 
     before(() => {
-        // # Testing Forwarding from Insights view requires a license
-        cy.apiRequireLicense();
-        cy.shouldHaveFeatureFlag('InsightsEnabled', true);
 
-        cy.apiUpdateConfig({
+        cy.apiUpdateConfig({Forwarding
```
[the remark this answers is not in the document]
Comment (marianunez) [reply e2e-tests/cypress/tests/integration/channels/message_forwarding/forward_message_from_dm_spec.js]: Typo - my bad. Removing

**Review by devinbinnie — APPROVED** [review]
Thanks for the work on this @harshilsharma63, approving for now to unblock but looks like a few tests are still broken. Happy to look again after those are fixed if you want me to.

File: webapp/platform/types/src/insights.ts (hunk 4 above)
Comment (harshilsharma63) [reply webapp/platform/types/src/insights.ts]: Hey yes, LOL is like my personal TODO, sorry for missing its removal before pushing.

**Review by yasserfaraazkhan — APPROVED** [review]
@harshilsharma63 I've checked on the spinwick, did not see insights showing up. I'm a little less aware of this feature. Please let me know if there are cases on desktop or mobile you'd suggest to test.

**Comment by harshilsharma63** [comment]
@yasserfaraazkhan I've tested and it look correct to me.

**Review by larkox — APPROVED** [review]

**Pull request merged**